include/eve/memory/soa_vector.hpp

```diff
@@ -0,0 +1,268 @@
+                    , storage, kumi::flatten_all(value)
+                    );
+    }
+
+    //! @brief Removes the last element of the container.
+    //! Calling pop_back on an empty container results in undefined behavior.
+    //! Iterators and references to the last element, as well as the end() iterator, are invalidated.
+    EVE_FORCEINLINE void pop_back() noexcept
+    {
+      kumi::for_each([&](auto& m) { m.pop_back(); }, storage);
+    }
+
+    //! @brief Resizes the container to contain count elements.
+    //!
+    //! If the current size is greater than `n`, the container is reduced to its first `n` elements.
+    //! If the current size is less than `n`, additional elements are appended. Those elements are
+    //! either default-inserted or copied if `value` is provided.
+    //! @param n      new size of the container
+    //! @param value  the value to initialize the new elements with
+    void resize(std::size_t n, value_type value = {})
+    {
+      kumi::for_each( [&](auto& m, auto const& v) { m.resize(n,v); }
+                    , storage, kumi::flatten_all(value)
+                    );
+    }
+
+    //! @brief Exchanges the contents of the container with those of `other`.
+    //! Does not invoke any move, copy, or swap operations on individual elements.
+    //! All iterators and references remain valid. The past-the-end iterator is invalidated.
+    void swap(soa_vector& other) noexcept
+    {
+      kumi::for_each([&](auto& m, auto& n) { m.swap(n); }, storage, other.storage);
+    }
+
+    //! @brief Swaps the contents of `lhs` and `rhs` by calling `lhs.swap(rhs)`.
+    friend EVE_FORCEINLINE void swap(soa_vector &lhs, soa_vector &rhs) noexcept
+    {
+      lhs.swap(rhs);
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    private:
+    template<typename T> static auto as_aligned_pointer(T* ptr)
+    {
+      return eve::as_aligned(ptr, detail::common_cardinal<T>{});
+    }
+
+    template<typename Self> static auto aligned_begin_impl(Self& self)
+    {
+       auto ptrs    = kumi::map([](auto& m) { return as_aligned_pointer(m.data()); }, self.storage);
+       auto zipped  = kumi::apply(eve::algo::zip, ptrs);
+
+       /* todo: add eve::algo::convert(zipped, eve::as<Type>{}) */
+       return zipped;
+    }
+
+    public:
+    //==============================================================================================
+    //! @name Element access
+    //! @{
+    //==============================================================================================
+    //! Returns an aligned pointer to the beginning
+    EVE_FORCEINLINE auto aligned_data()  { return aligned_begin_impl(*this); }
+
+    //! Returns an aligned pointer to the beginning
+    EVE_FORCEINLINE auto aligned_data()  const { return aligned_begin_impl(*this); }
```

Comment:
Stepanov advocated that the `what` should come first.
Like `stable_sort` is a bad name and it should be `sort_stable` because of an alphabetical listing.

Do you want to change to `data_aligned`? It's a suggestion, up to you

Reply:
I'm ok with both so le'ts do that